OpenKeychain/src/main/java/org/sufficientlysecure/keychain/service/PassphraseCacheService.java

```diff
@@ -314,7 +314,8 @@ public void onReceive(Context context, Intent intent) {
     private static PendingIntent buildIntent(Context context, long keyId) {
         Intent intent = new Intent(BROADCAST_ACTION_PASSPHRASE_CACHE_SERVICE);
         intent.putExtra(EXTRA_KEY_ID, keyId);
-        return PendingIntent.getBroadcast(context, REQUEST_ID, intent,
+        Long _id = System.currentTimeMillis();
+        return PendingIntent.getBroadcast(context, _id.intValue(), intent,
```

Comment:
Sir actually pending indent requires request-code so what happens is when two pending indents query with same request code then one which is later gets the response or i mean to say the later one is deleted from cache but not the former one because the request code has been overwritten as request code was harcoded to 0.